src/libs/ttf/lv_ttf.h

```diff
@@ -0,0 +1,48 @@
+#ifndef LV_TTF_H
+#define LV_TTF_H
+#ifdef __cplusplus
+extern "C" {
+#endif
+#include "../../../lvgl.h"
+// private structure
+typedef struct lv_ttf_buf {
+    lv_fs_file_t *stream;
+    size_t offset;
+    int cursor;
+    int size;
+} lv_ttf_buf_t;
+
+typedef struct lv_ttf {
+    void *userdata;
+    //unsigned char *data; // pointer to .ttf file
+    lv_fs_file_t *stream; // stream of .ttf file
```

Comment:
```suggestion
    lv_fs_file_t f; 
```

Open the file in `lv_ttf_open` to `f`.

Reply:
That concerns me. Keep in mind, if you do that you are making the TTF engine responsible for the lifetime of the file (unless I am wrong)

If you're okay with that, that's fine, but it require closing and opening the file somewhere in the code, and not when you create the font, because several lv_font_t instances can be created from a single open file.